transformer_engine/common/fused_attn/fused_attn_fp16_bf16_max_seqlen_512.cu

```diff
@@ -668,7 +667,8 @@ void fused_attn_max_512_fwd_impl(int64_t b, int64_t h, int64_t s_q, int64_t s_kv
             createScale(b, h, s_q, s_kv, d, layout, tensorType, ops);
 
             // if bias, we need to memset the S buffer to correctly computate dbias
-            auto zero_s = (bias_type != NVTE_Bias_Type::NVTE_NO_BIAS);
+            auto zero_s = (bias_type != NVTE_Bias_Type::NVTE_NO_BIAS) ||
+                          (mask_type == NVTE_Mask_Type::NVTE_CAUSAL_MASK);
```

Comment:
I'd like to understand this bugfix a bit more. The cuDNN bug only shows up with `causal_mask` and not the other mask modes?

Reply:
Yes, see `nvbugs 4122719` for details.